autoplex/data/rss/utils.py

```diff
@@ -305,49 +322,93 @@ def __repr__(self):
-    mlip_path,
-    output_file_name,
-    mlip_type,
-    scalar_pressure_method,
-    scalar_exp_pressure,
-    scalar_pressure_exponential_width,
-    scalar_pressure_low,
-    scalar_pressure_high,
-    max_steps,
-    force_tol,
-    stress_tol,
-    Hookean_repul,
-    hookean_paras,
-    write_traj,
-    device,
-    isol_es,
-):
-    """Run RSS on a single thread using MLIPs."""
+    atom: Atoms,
+    mlip_type: str,
+    mlip_path: str,
+    output_file_name: str = "RSS_relax_results",
+    scalar_pressure_method: str = "exp",
+    scalar_exp_pressure: float = 100,
+    scalar_pressure_exponential_width: float = 0.2,
+    scalar_pressure_low: float = 0,
+    scalar_pressure_high: float = 50,
+    max_steps: int = 1000,
+    force_tol: float = 0.01,
+    stress_tol: float = 0.01,
+    hookean_repul: bool = False,
+    hookean_paras: dict | None = None,
+    write_traj: bool = True,
+    device: str = "cpu",
+    isolated_atom_energies: dict[int, float] | None = None,
+    config_type: str = "traj",
+    keep_symmetry: bool = True,
+) -> str | None:
+    """Run RSS on a single thread using MLIPs.
+
+    Parameters
+    ----------
+    atom: Atoms
+        ASE Atoms object representing the atomic configuration.
+    mlip_type: str
+        Choose one specific MLIP type:
+        'GAP' | 'J-ACE' | 'P-ACE' | 'NequIP' | 'M3GNet' | 'MACE'.
+    mlip_path: str
+        Path to the MLIP model.
+    output_file_name: str
+        Prefix for the trajectory/log file name. The actual output file name
+        may be composed of this prefix, an index, and file types.
+    scalar_pressure_method: str
+        Method for adding external pressures. Default is 'exp'.
+    scalar_exp_pressure: float
+        Scalar exponential pressure. Default is 100.
+    scalar_pressure_exponential_width: float
+        Width for scalar pressure exponential. Default is 0.2.
+    scalar_pressure_low: float
+        Low limit for scalar pressure. Default is 0.
+    scalar_pressure_high: float
+        High limit for scalar pressure. Default is 50.
+    max_steps: int
+        Maximum number of steps for relaxation. Default is 1000.
+    force_tol: float
+        Force residual tolerance for relaxation. Default is 0.01.
+    stress_tol: float
+        Stress residual tolerance for relaxation. Default is 0.01.
+    hookean_repul: bool
+        If true, apply Hookean repulsion. Default is False.
+    hookean_paras: dict[tuple[int, int], tuple[float, float]]
+        Parameters for Hookean repulsion as a dictionary of tuples. Default is None.
+    write_traj: bool
+        If true, write trajectory of RSS. Default is True.
+    device: str
+        Specify device to use "cuda" or "cpu".
+    isolated_atom_energies: dict
+        Dictionary of isolated atoms energies.
+    config_type: str
+        Specify the type of configurations generated from RSS.
+    keep_symmetry: bool
+        If true, preserve symmetry during relaxation.
+
+    Returns
+    -------
+    str | None
```

Comment:
Maybe keep here only str as returns

And add another header `Raises`, and under that just mention It will raise RuntimeError when optimization fails?

Reply:
No, here we should use `str | None`. Since we've set a maximum number of relaxation steps, for structures far from equilibrium, even if they haven't relaxed successfully, we can still accept them. When sampling, we will discard all those marked as `None`.